### PR TITLE
CFY-6009 Add the HA cluster client

### PR DESCRIPTION
**Pull request by tehasdf**

### CODE REVIEW

File: cloudify_rest_client/cluster.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+class ClusterState(dict):
+    @property
+    def initialized(self):
+        return self.get('initialized', False)
+
+    @property
+    def encryption_key(self):
+        try:
```
Comment (dusking) [inline cloudify_rest_client/cluster.py]: why not:
return self.get('encryption_key')

File: cloudify_rest_client/cluster.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    @property
+    def error(self):
+        return self.get('error')
+
+
+class ClusterClient(object):
+    def __init__(self, api):
+        self.api = api
+        self.nodes = ClusterNodesClient(api)
```
Comment (dusking) [inline cloudify_rest_client/cluster.py]: where are we using self.nodes?

File: cloudify_rest_client/cluster.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        :param config: new cluster configuration
+        :return: current cluster status
+        """
+        response = self.api.patch('/cluster', data={
+            'config': config
+        })
+        return ClusterState(response)
+
+
+class ClusterNodesClient(object):
```
Comment (dusking) [inline cloudify_rest_client/cluster.py]: are we **using** ClusterNodesClient?

File: cloudify_rest_client/cluster.py (hunk 1 above)
Comment (tehasdf) [inline cloudify_rest_client/cluster.py]: whoops! it used to be nested before so that was cleaner. I removed the nesting but forgot to replace this :)

**Pull request merged**